distribution/lib/Standard/Table/0.0.0-dev/src/Data/Group_By.enso

```diff
@@ -0,0 +1,25 @@
+from Standard.Base import all
+
+## Group By Key
+type Group_By_Key
+    type Group_By_Key values
+
+    ## See if two keys are equal
+    == : Group_By_Key->Boolean
+    == that =
+        if that.values.length != this.values.length then False else
+            0.up_to this.values.length . all i->(this.values.unsafe_at i == that.values.unsafe_at i)
+
+    ## See if two keys are not equal
+    != : Group_By_Key->Boolean
+    != that = this==that . not
+
+    ## Compares two keys
+    compare_to : Group_By_Key->Ordering
+    compare_to that =
+        loop i =
+            comparison = this.values.unsafe_at i . compare_to (that.values.unsafe_at i)
+            if comparison != Ordering.Equal then comparison else
+                if i == this.values.length - 1 then Ordering.Equal else
+                    @Tail_Call loop (i+1)
+        loop 0
```

Comment:
Again, cannot we just reuse `Vector_Lexicographic_Order.compare`? Seems like this is the same algorithm modulo the fact that this algorithm does not check if both vectors have the same length so it can crash in a nasty way (as it uses `unsafe_at`) if they do not. I know normally we assume the group keys will be of the same length, but IMO if using `unsafe_at` the check should be there (and interestingly, it is there for `==`).